osu.Game/Utils/NamingUtils.cs

```diff
@@ -57,5 +58,19 @@ public static string GetNextBestName(IEnumerable<string> existingNames, string d
                 ? desiredName
                 : $"{desiredName} ({bestNumber})";
         }
+
+        /// <summary>
+        /// Given a set of <paramref name="existingFilenames"/> and a desired target <paramref name="desiredName"/>
+        /// finds a filename closest to <paramref name="desiredName"/> that is not in <paramref name="existingFilenames"/>
+        /// <remarks>
+        /// <paramref name="desiredName"/> SHOULD NOT CONTAIN the file extension.
+        /// </remarks>
+        /// </summary>
+        public static string GetNextBestFilename(IEnumerable<string> existingFilenames, string desiredName, string fileExtension)
```

Comment:
Yeah I'm not a fan of this method's params as they are currently. The extension should arguably be part of the `desiredName` for convenience. Not to mention that due to the split the current implementation of this method has a bug, wherein if you call
```csharp
GetNextBestFilename(new[] { "test.jpg" }, "test", "png");
```
it will actually return `test (1).png` because of the stripping of extensions from `existingFilenames` done below. Needs test coverage for sure (see `NamingUtilsTest`).